src/registrar/tests/test_views.py

```diff
@@ -334,3 +309,69 @@ def test_application_form_submission(self):
         self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
         final_result = review_result.follow()
         self.assertContains(final_result, "Thank you for your domain request")
+
+    def test_application_form_conditional_federal(self):
+        """Federal branch question is shown for federal organizations."""
+        type_page = self.app.get(reverse("application")).follow()
+        # django-webtest does not handle cookie-based sessions well because it keeps
+        # resetting the session key on each new request, thus destroying the concept
+        # of a "session". We are going to do it manually, saving the session ID here
+        # and then setting the cookie on each request.
+        session_id = self.app.cookies[settings.SESSION_COOKIE_NAME]
+
+        # ---- TYPE PAGE  ----
+
+        # the conditional step titles shouldn't appear initially
+        self.assertNotContains(type_page, TITLES["organization_federal"])
+        self.assertNotContains(type_page, TITLES["organization_election"])
+        type_form = type_page.form
+        type_form["organization_type-organization_type"] = "Federal"
```

Comment:
Great job with the unit tests; I know the session id was a major pain, so kudos for sticking with it.